src/OrbitGl/OpenGlBatcher.h

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_OPEN_GL_BATCHER_H_
+#define ORBIT_GL_OPEN_GL_BATCHER_H_
+
+#include <Batcher.h>
+
+#include <array>
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+#include "Containers/BlockChain.h"
+#include "PickingManager.h"
+
+namespace orbit_gl {
+
+namespace {
+
+struct LineBuffer {
+  void Reset() {
+    lines_.Reset();
+    colors_.Reset();
+    picking_colors_.Reset();
+  }
+
+  static const int NUM_LINES_PER_BLOCK = 64 * 1024;
+  orbit_containers::BlockChain<Line, NUM_LINES_PER_BLOCK> lines_;
+  orbit_containers::BlockChain<Color, 2 * NUM_LINES_PER_BLOCK> colors_;
+  orbit_containers::BlockChain<Color, 2 * NUM_LINES_PER_BLOCK> picking_colors_;
+};
+
+struct BoxBuffer {
+  void Reset() {
+    boxes_.Reset();
+    colors_.Reset();
+    picking_colors_.Reset();
+  }
+
+  static const int NUM_BOXES_PER_BLOCK = 64 * 1024;
+  orbit_containers::BlockChain<Box, NUM_BOXES_PER_BLOCK> boxes_;
+  orbit_containers::BlockChain<Color, 4 * NUM_BOXES_PER_BLOCK> colors_;
+  orbit_containers::BlockChain<Color, 4 * NUM_BOXES_PER_BLOCK> picking_colors_;
+};
+
+struct TriangleBuffer {
+  void Reset() {
+    triangles_.Reset();
+    colors_.Reset();
+    picking_colors_.Reset();
+  }
+
+  static const int NUM_TRIANGLES_PER_BLOCK = 64 * 1024;
+  orbit_containers::BlockChain<Triangle, NUM_TRIANGLES_PER_BLOCK> triangles_;
+  orbit_containers::BlockChain<Color, 3 * NUM_TRIANGLES_PER_BLOCK> colors_;
+  orbit_containers::BlockChain<Color, 3 * NUM_TRIANGLES_PER_BLOCK> picking_colors_;
+};
+
+struct PrimitiveBuffers {
+  void Reset() {
+    line_buffer.Reset();
+    box_buffer.Reset();
+    triangle_buffer.Reset();
+  }
+
+  LineBuffer line_buffer;
+  BoxBuffer box_buffer;
+  TriangleBuffer triangle_buffer;
+};
+
+}  // namespace
+/**
+Implements internal methods to collects primitives to be rendered at a later point in time.
+
+NOTE: The OpenGlBatcher assumes x/y coordinates are in pixels and will automatically round those
+down to the next integer in all Batcher::AddXXX methods. This fixes the issue of primitives
+"jumping" around when their coordinates are changed slightly.
+**/
```

Comment:
Nit: We tend to use `//` instead of `/* */`

Reply:
Also an empty line above the comment would be great

Reply:
Yes, it was copy pasted directly from Batcher.